qiskit/circuit/equivalence.py

```diff
@@ -166,7 +215,7 @@ def _build_basis_graph(self):
         graph = rx.PyDiGraph()
 
         node_map = {}
-        for key in self._get_all_keys():
+        for key in self._key_to_node_index.keys():
```

Comment:
You don't really need `keys()` here it just creates an intermediate view object, it'll be marginally faster to just iterate over the dict directly:

```suggestion
        for key in self._key_to_node_index:
```

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/8790/commits/837b18558b075631bed822f4c3ba2cee13b98b87.